compiler/rustc_lint/src/early.rs

```diff
@@ -426,7 +426,7 @@ pub fn check_ast_node<'a>(
     if !sess.opts.debugging_opts.no_interleave_lints {
```

Comment:
not part of this PR, but can you swap these two branches so that the `if` has a positive condition?